### DISCUSSION TITLE
Dynamic Component Name In Templates

### DISCUSSION THREAD

**Discussion by impeto** [livewire/livewire#3155] [1. Help] [answered]
So I have this scenario. I have a `Blade` component `dashboard` with these contents:

```php
@props(['component', 'title'])

<x-app-layout>
    <x-slot name="header">
        <h2 class="font-semibold text-xl text-gray-800 leading-tight">
            {{ $title }}
        </h2>
    </x-slot>

    @livewire({{ $component }})
</x-app-layout>
```

Then I have `views/dashboards/dashboard.blade.php` that contains this:

```php
<x-dashboard :component="$component" :title="$title" />

```

When I call the `dashboards.dashboard` view from a controller method like this:

```php
        return view('dashboards.dashboard', [
            'component' => 'accounts.view-account',
            'title' => 'My Account'
        ]);
```

I get this error:

```
ParseError
syntax error, unexpected token "<" (View: [...]/resources/views/components/dashboard.blade.php)
```

This is due to line 45 in `Livewire\LivewireBladeDirectives.php`

```php
\$html = \Livewire\Livewire::mount({$expression})->html();
```
which gets translated into this
```php
$html = \Livewire\Livewire::mount(<?php echo e($component); ?>)->html();
```
after parsing. Am I not supposed to be able to do that in LW or is it some kind of a bug?

**Comment by joshhanley** [comment, marked as answer]
@impeto you can't use a blade `@` directive with blade echo `{{ }}`, so try change this 
```blade
 @livewire({{ $component }})
 ```
 
 to this (I've removed the `{{ }}` from around `$component`)
 ```blade
 @livewire($component)
 ```
 
 Hope this helps!

**Reply by impeto** [reply]
Thank you for responding. I had tried both ways. It turns out that's only part of it. The other part is the variable name `$component`. I believe `$component` in that context is `x-app-layout`. I changed the variable name to `$comp` and it works. I knew it had to be something stupid like that.  Need more sleep :) Thanks again!

**Reply by joshhanley** [reply]
@impeto no worries 🙂 

Ah yep, `$component` is used inside blade components for internal stuff, so that makes sense. I had forgotten about that!

Glad it's sorted. And get some sleep! 😆
